src/moonlink/src/table_handler.rs

```diff
@@ -194,29 +195,40 @@ impl TableHandler {
                         // Replication events
                         // ==============================
                         //
-                        TableEvent::Append { row, xact_id } => {
-                            let result = match xact_id {
-                                Some(xact_id) => {
-                                    let res = table.append_in_stream_batch(row, xact_id);
-                                    if table.should_transaction_flush(xact_id) {
-                                        if let Err(e) = table.flush_transaction_stream(xact_id).await {
-                                            error!(error = %e, "flush failed in append");
-                                        }
+                        TableEvent::Append { is_copied, row, xact_id } => {
+                            if is_copied || (!table.is_in_initial_copy() && xact_id.is_none()) {
+                                if let Err(e) = table.append(row) {
+                                    warn!(error = %e, "failed to append row");
+                                }
+                                if table.should_flush() {
```

Comment:
Why do we keep flushing with LSN 0 ??
It will cause all later snapshot to have LSN 0, for non initial copy snapshots

Reply:
To make it worse, flush LSN is recorded as data file flush LSN, which would regress after this PR

Reply:
I think for flush there're two invariants we should keep:
- Flush when the table is at a clean state
- Flush LSN never regresses

Reply:
And for the current implementation, flushing in the middle of initial copy also doesn't work, because snapshot logic assume "only flush at a clean state".

In detail, when persist to iceberg, we only check flush LSN (and assume it's committed). So it would lead to partial write to iceberg, which puts our persistence layer in an inconsistent state.

Reply:
I plan to remove the whole flush logic here for both initial copy and normal non-streaming write.

If you want periodical flush, why not using streaming write buffer?